From 87794f1a864f3681343bc764fdd98d746ecbee57 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Wed, 27 Mar 2024 13:44:41 +0800
Subject: [PATCH] common_defs.h: improve PowerPC detection macros

This adds macros to support macOS, AIX and 32-bit BSD.
May not be exhaustive, but should cover most systems, hopefully.
---
 common_defs.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git common_defs.h common_defs.h
index 094d438..48e7547 100644
--- common_defs.h
+++ common_defs.h
@@ -401,8 +401,9 @@ static forceinline u64 bswap64(u64 v)
  */
 #if (defined(__GNUC__) || defined(__clang__)) && \
 	(defined(ARCH_X86_64) || defined(ARCH_X86_32) || \
-	 defined(__ARM_FEATURE_UNALIGNED) || defined(__powerpc64__) || \
-	 defined(__riscv_misaligned_fast) || \
+	 defined(__ARM_FEATURE_UNALIGNED) || defined(__PPC) || \
+	 defined(__powerpc__) || defined(__powerpc64__) || \
+	 defined(__POWERPC__) || defined(__riscv_misaligned_fast) || \
 	 /*
 	  * For all compilation purposes, WebAssembly behaves like any other CPU
 	  * instruction set. Even though WebAssembly engine might be running on
